lib/fetchContent.js

```diff
@@ -348,6 +348,227 @@ function getArticleRelations(days, facets, aspects)
 		});
 }
 
+function getArticlesAggregation(days, facets, aspects, minCorrelation=2)
+{
+	const date = time.getDatetimeRange('days', days, 0);
+	aspects = checkMetaDataPresent(aspects);
+	const params = {
+		queryString: `lastPublishDateTime:> ${date.next}`,
+		maxResults : 100,
+		aspects : aspects,
+		facets : {
+			"names": facets,
+			"maxElements":-1
+		}
+	};
+
+	return searchDeeper(params, 10)
+		.then(results =>  extractArticles(results) )
+		.then(articles => aggregateArticles(articles, facets, minCorrelation, days, params, filterByMetadataCsv=null) )
+		.then(aggregation => {
+			// look for all the distinct genre CSVs
+			let allCsvs = Object.keys(aggregation.articlesByMetadataCsv);
+			let allGenreCsvs = allCsvs.filter( csv => csv.startsWith('genre:') );
+			// re-establish the full list of articles
+			let articles = Object.values(aggregation.articlesByUuid);
+			// construct an aggregation for each genre csv
+			let aggregationsByGenre = {};
+			allGenreCsvs.forEach( csv => {
+				aggregationsByGenre[csv] = aggregateArticles(articles, facets, minCorrelation, days, params, filterByMetadataCsv=csv);
+			});
+			let articleCounts = {};
+			allGenreCsvs.forEach(csv => {
+				articleCounts[csv] = aggregationsByGenre[csv].numArticles;
+			});
+
+			return {
+				'description' :[
+					'calculating aggregations for each genre CSV',
+				],
+				articleCounts,
+				aggregationsByGenre
+			};
+		})
+		.catch(err => {
+			throw err;
+		});
+}
+
+function addUuidToMetadataTerm(grouping, metadataKey, term, uuid){
+	if (!grouping.hasOwnProperty(metadataKey)) {
+		grouping[metadataKey] = {};
+	}
+
+	if (!grouping[metadataKey].hasOwnProperty(term.taxonomy)) {
+		grouping[metadataKey][term.taxonomy] = {};
+	}
+
+	if (!grouping[metadataKey][term.taxonomy].hasOwnProperty(term.name)) {
+		grouping[metadataKey][term.taxonomy][term.name] = [];
+	}
+
+	grouping[metadataKey][term.taxonomy][term.name].push(uuid);
+}
+
+function addFacetCorrelation( grouping, metadataKey1, term1, metadataKey2, term2 ){
+	if (!grouping.hasOwnProperty(metadataKey1)) {
+		grouping[metadataKey1] = {};
+	}
+
+	if (!grouping[metadataKey1].hasOwnProperty(term1.taxonomy)) {
+		grouping[metadataKey1][term1.taxonomy] = {};
```

Comment:
`grouping[metadataKey1][term1].taxonomy`. Is `term1` always defined?

Reply:
yes.
refactored.